src/util.typ

```diff
@@ -15,12 +15,16 @@
 
 /// Multiplies vectors by the transform matrix
 ///
-/// - transform (matrix): Transformation matrix
+/// - transform (matrix,function): Transformation matrix or a function that accepts and returns a vector.
 /// - ..vecs (vectors): Vectors to get transformed. Only the positional part of the sink is used. A dictionary of vectors can be passed and all will be transformed.
 /// -> vectors If multiple vectors are given they are returned as an array, if only one vector is given only one will be returned, if a dictionary is given they will be returned in the dictionary with the same keys.
 #let apply-transform(transform, ..vecs) = {
-  let t = vec => matrix.mul4x4-vec3(
-    transform, vec)
+  let t = if type(transform) != function {
+    vec => matrix.mul4x4-vec3(
```

Comment:
idk how long its been like this but use `matrix.mul4x4-vec3.with(transform)` instead